src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -665,6 +666,18 @@ public PossiblyConditionalState Clone()
             nodeStateMap.Free();
             return labelStateMap;
 
+            void ignoreAnnotationAndReport(int outputSlot, TypeWithAnnotations outputType, ref LocalState state, BoundDagSliceEvaluation e)
+            {
+                if (PossiblyNullableType(outputType.Type))
+                {
+                    if (state[outputSlot] == NullableFlowState.MaybeNull)
+                    {
+                        ReportDiagnostic(ErrorCode.WRN_AnnotationOnSliceReturnType, e.Syntax, outputType);
+                        state[outputSlot] = NullableFlowState.NotNull;
+                    }
+                }
+            }
```

Comment:
I'm not entirely sure about this check. @RikkiGibson is this the right way do this?

Reply:
This check can be removed after all (not necessary). Thanks

Reply:
Will do. I didn't understand what's the rationale though. Some of warnings may end up being nonsensical. Is that ok?